src/chai-CopyCode.ts

```diff
@@ -0,0 +1,219 @@
+  @property({type: Boolean}) showAddress = false;
+
+  private observer: MutationObserver | null = null;
+
+  override updated(changedProperties: Map<string, any>) {
+    super.updated(changedProperties);
+    if (changedProperties.has('targetId') && this.targetId) {
+      this.requestTargetElement();
+    }
+
+    if (
+      changedProperties.has('showName') ||
+      changedProperties.has('showPhone') ||
+      changedProperties.has('showEmail') ||
+      changedProperties.has('showAddress')
+    ) {
+      this.initializeDefaultValues();
+    }
+  }
+
+  private requestTargetElement() {
+    this.dispatchEvent(
+      new CustomEvent('request-target-element', {
+        bubbles: true,
+        composed: true,
+        detail: {
+          targetId: this.targetId,
+          callback: this.setTargetElement.bind(this),
+        },
+      })
+    );
+  }
+
+  private setTargetElement(element: HTMLElement | null) {
+    this.targetElement = element;
+    if (this.targetElement) {
+      this.initializeDefaultValues();
+      this.observeTargetElement();
+    } else {
+      console.warn(`Element with id "${this.targetId}" not found`);
+    }
+  }
+
+  private initializeDefaultValues() {
+    if (this.targetElement) {
+      const styles = this.targetElement.getAttribute('style');
+      this.cssValues = `
+${
+  styles
+    ? `
+<style>
+  chai-form${
+    this.targetElement.className ? `.${this.targetElement.className}` : ''
+  }{
+    ${styles.replaceAll(';', '; \n')}
+  }
+</style>`
+    : ''
+}
+<chai-form${
+        this.targetElement.className
+          ? ` class="${this.targetElement.className}"`
+          : ''
+      }${
+        this.targetElement.getAttribute('headertext')
+          ? ` headertext="${this.targetElement.getAttribute('headertext')}"`
+          : ''
+      }${
+        this.targetElement.getAttribute('buttontext')
+          ? ` buttontext="${this.targetElement.getAttribute('buttontext')}"`
+          : ''
+      }>
+    ${this.showName ? `<chai-name></chai-name>` : ''}
+    ${this.showEmail ? `<chai-email></chai-email>` : ''}
+    ${this.showPhone ? `<chai-phone></chai-phone>` : ''}
+    ${this.showAddress ? `<chai-address></chai-address>` : ''}
+</chai-form>`.trim();
+    }
+  }
+
+  private observeTargetElement() {
+    if (this.targetElement) {
+      this.observer = new MutationObserver(() => {
+        this.initializeDefaultValues();
+      });
+
+      this.observer.observe(this.targetElement, {
+        attributes: true,
+        attributeFilter: ['style', 'class', 'headertext', 'buttontext'],
+      });
+    }
+  }
+
+  override disconnectedCallback() {
+    if (this.observer) {
+      this.observer.disconnect();
+    }
+    super.disconnectedCallback();
+  }
+
+  override render() {
+    return html`
+      <div class="form-control-container">
+        <div class="form-control-header">
+          <button @click=${this.copyToClipboard}>Copy CSS Variables</button>
```

Comment:
The `<style>` and `<chai-form>` should be separate; one usually goes in the website header, while the other goes wherever it needs to on the page.